x/market/handler/endblock.go

```diff
@@ -55,6 +56,49 @@ func transferFundsForActiveLeases(ctx sdk.Context, keepers Keepers) error {
 	return nil
 }
 
+func pickBidWinner(bids []types.Bid) (winner types.Bid) {
+	// open bids; match by lowest price; sort bids by price
+	sort.Slice(bids, func(i, j int) bool {
+		// AFAICT: the BidID DSeq is pulled from the original OrderID.
+		// So it can't be used to determine who bid first.
+		return bids[i].Price.IsLT(bids[j].Price)
+	})
+	if len(bids) == 0 {
+		// This is a fatal case
+		winner = types.Bid{}
+		return
+	} else if len(bids) == 1 {
+		winner = bids[0]
+		return
+	} // else > 2 bids
+
+	if bids[0].Price.IsEqual(bids[1].Price) {
+		// There are equivalent bid prices; select winner with deterministic
+		// random ordering based on given bids.
+		equalBids := make([]types.Bid, 0)
+		for i := 0; bids[0].Price.IsEqual(bids[i].Price); i++ {
+			equalBids = append(equalBids, bids[i])
```

Comment:
might be small performance gain by just finding the index and using `bids[:idx]` or even just `idx` later.

Reply:
That might help solve the simulations index error, good idea.